File: backend/models/others.go
// Autores:
//   - Jennyfer Jasso

package models

// Estructura de la tabla de otros (otra información del cliente)
type Other struct {
	ID        uint   `json:"-" gorm:"primarykey"`
	CURP      string `json:"curp" validate:"required,len=18"`
	Street    string `json:"street" validate:"required"`
	Interior  *int   `json:"interior" validate:"omitempty"`
	Exterior  int    `json:"exterior"`
	City      string `json:"city" validate:"required,min=1"`
	State     string `json:"state" validate:"required,min=1"`
	ZIP       string `json:"zip" validate:"required,min=4,max=10"`
	Reference string `json:"reference"`
	ClientID  uint   `json:"-"`
}

File: app/app/src/main/java/mx/cazv/todasbrillamos/model/models/Others.kt
package mx.cazv.todasbrillamos.model.models

/**
 * Data class con la información adicional de un cliente.
 * @author Carlos Zamudio
 *
 * @property CURP El CURP del cliente.
 * @property Street La calle donde vive el cliente.
 * @property Interior El número interior de la dirección del cliente.
 * @property Exterior El número exterior de la dirección del cliente.
 * @property City La ciudad donde vive el cliente.
 * @property State El estado donde vive el cliente.
 * @property ZIP El código postal del cliente.
 * @property Reference Una referencia adicional (punto de referencia) de la dirección del cliente.
 */
data class Others(
    val CURP: String,
    val Street: String,
    val Interior: String,
    val Exterior: String,
    val City: String,
    val State: String,
    val ZIP: String,
    val Reference: String
)